contracd/contracd.pro

```diff
@@ -4,6 +4,13 @@ VERSION_MAJOR = 0
 VERSION_MINOR = 0
 VERSION_BUILD = 1
 
+#protobuf build step
+PRE_TARGETDEPS += $$PWD/src/contrac.pb.cc
+protobuf.target = $$PWD/src/contrac.pb.cc
+protobuf.commands = protoc --proto_path=$$PWD/src --cpp_out=$$PWD/src $$PWD/src/contrac.proto
```

Comment:
Looks good. Except this mixes generated and non-generated code.

Reply:
I've changed the process to move the generation into the spec file, which I think is cleaner. Hopefully this address your comment here too.

Reply:
Thats not what I meant this still belongs into the qmake file but the destination of the generated source should be $$OUT_PWD[1] and not $$PWD. $$OUT_PWD is the place where the generated Makefile lies and thous the buildir.

[1] https://doc.qt.io/qt-5/qmake-variable-reference.html#out-pwd

A more complicated solution is this: https://wiki.qt.io/QMake-top-level-srcdir-and-builddir

Reply:
Thanks for the info, that's helpful. I've amended it so that the generated files could placed in the build directory, rather than the source tree. This does feel more correct and seems to work well. I feel like this ought to be going in the prepare step, but that doesn't seem to be supported by by the application SDK, so this is a good alternative.

Reply:
Thats normal prepare is for changes outside of the sources like patches. The build process shouldn't depend on the packaging system.

Reply:
Like I say, I'm not entirely convinced this generation should be part of the build process, but it seems convenient under the circumstances.